libparsec/crates/types/src/certif.rs

```diff
@@ -1001,13 +1001,32 @@ impl_unsecure_dump!(ShamirRecoveryBriefCertificate);
 impl_dump_and_sign!(ShamirRecoveryBriefCertificate);
 
 impl ShamirRecoveryBriefCertificate {
+    pub fn check_data_integrity(&self) -> DataResult<()> {
+        let total_shares = {
+            let mut total_shares: u64 = 0;
+            for shares in self.per_recipient_shares.values() {
+                total_shares += u64::from(*shares);
+            }
+            total_shares
+        };
+        if u64::from(self.threshold) > total_shares {
+            return Err(DataError::DataIntegrity {
+                data_type: std::any::type_name::<Self>(),
+                invariant: "threshold <= total_shares",
+            });
+        }
+        Ok(())
+    }
+
     pub fn verify_and_load(
         signed: &[u8],
         author_verify_key: &VerifyKey,
         expected_author: DeviceID,
     ) -> DataResult<Self> {
         let r = verify_and_load::<Self>(signed, author_verify_key)?;
 
+        r.check_data_integrity()?;
```

Comment:
I find it logical to have this check here.

But then there is no need to also have it in the `add_certificates_batch`. This is especially true given `add_certificates_batch` receive the certificates as `Bytes` and hence deserializes them itself (vs if `ShamirRecoveryBriefCertificate` object were provided to it and it then had to trust the caller on the fact those objects hasn't been manually modified after creation/deserialization)

On top of that, there is the remote manifests that now have a `check_data_integrity` method, but which is not called in their  `verify_and_load` method...

So my guess:
- Having a *public* `check_data_integrity` method is useful to plug on object that get modified across they lifetime (so only local manifest).
- All types should call their own `check_data_integrity` when deserializing (i.e. the "verify_and_load" means "verify the signature, verify the format by deserializing, and verify the integrity")
- We *may* want to add an assert on a call to `check_data_integrity` before each serialization as a sanity check.
- `FolderManifest` should expose `FolderManifest::verify_and_load_for_root` and `FolderManifest::verify_and_load_for_child` instead of a single `FolderManifest::verify_and_load`